clippy_lints/src/utils/author.rs

```diff
@@ -400,9 +475,29 @@ impl<'tcx> Visitor<'tcx> for PrintVisitor {
                     self.visit_pat(arm.pat);
                 }
             },
-            ExprKind::Closure(ref _capture_clause, _func, _, _, _) => {
-                println!("Closure(ref capture_clause, ref func, _, _, _) = {};", current);
-                println!("    // unimplemented: `ExprKind::Closure` is not further destructured at the moment");
+            ExprKind::Closure(ref capture_clause, _, body, _, _) => {
+                let capture_by_pat = self.next("capture_by");
+                let async_pat = self.next("async");
+                let fn_expr_pat = self.next("fn_expr");
+
+                println!(
+                    "Closure(ref {}, ref {}, _, _, ref {}, _) = {};",
+                    capture_by_pat, async_pat, fn_expr_pat, current
+                );
+
+                println!("    if let CaptureBy::{:?} = {}", capture_clause, capture_by_pat);
+
+                let hir = self.cx.unwrap().tcx.hir();
+                let body = hir.body(body);
+
+                if let Some(GeneratorKind::Async(AsyncGeneratorKind::Closure)) = body.generator_kind {
```

Comment:
I think no because the generator is something that gets desugared alongside the body that is directly from code.

Use `-Zunpretty=hir-tree` to see what the HIR is for any given code.

Reply:
Thanks, that flag makes things a lot easier.